lib/kubeclient/common.rb

```diff
@@ -313,8 +317,12 @@ def create_faraday_client(url = nil)
         if @auth_options[:username] && @auth_options[:password]
           connection.basic_auth(@auth_options[:username], @auth_options[:password])
         end
-        connection.use(FaradayMiddleware::FollowRedirects, limit: @http_max_redirects)
-        connection.response(:raise_error)
+        if block_given?
+          yield(connection)
+        else
+          connection.use(FaradayMiddleware::FollowRedirects, limit: @http_max_redirects)
+          connection.response(:raise_error)
+        end
```

Comment:
It might just be easier to slip the block in here and _always_ use lines 323, 324; that might actually be the behaviour people would expect, now that I think about it....

Reply:
Ok, I've changed this to always include `:raise_error` and `FollowRedirects`. I think those assumptions are baked into the client enough to justify their forced inclusion (I'll be sure to document that in the README, mind you)